server/src/WebsocketServer.hs

```diff
@@ -104,9 +104,14 @@ handleClient conn path core = do
       currentValue <- getCurrentValue core path
       log (T.pack $ "Sending initial value to client: " ++ show currentValue) logRecords
       WS.sendTextData conn (Aeson.encode currentValue)
+
+    closeHandshake :: WS.ConnectionException -> IO ()
+    closeHandshake (WS.CloseRequest _ _) = WS.sendClose conn ("" :: Text)
```

Comment:
I would add that the websocket protocol requires sending an empty frame as part of the closing handshake.